model/convert/service_event.go

```diff
@@ -395,9 +395,34 @@ func convertClusterQCVotes(cdcClusterQCs []cadence.Value) ([]flow.ClusterQCVoteD
 		// Aggregate BLS signatures
 		aggregatedSignature, err := crypto.AggregateBLSSignatures(signatures)
 		if err != nil {
+			// expected errors of the function are:
+			//  - empty list of signatures
+			//  - an input signature does not deserialize to a valid point
+			// Both are not expected at this stage because list is guaranteed not to be
+			// empty and individual signatures have been validated.
 			return nil, fmt.Errorf("cluster qc vote aggregation failed: %w", err)
 		}
 
+		// check that aggregated signature is not identity, because an identity signature
+		// is invalid if verified under an identity public key. This can happen in two cases:
+		//  - If the quorum has at least one honest signer, and given all staking key proofs of possession
+		//    are valid, it's extremely unlikely for the aggregated public key (and the corresponding
+		//    aggregated signature) to be identity.
+		//  - If all quorum is malicious and intentionally forge an identity aggregate. As of the previous point,
+		//    this is only possible if there is no honest collector involved in constructing the cluster QC.
+		//    Hence, the cluster would need to contain a supermajority of malicious collectors.
+		//    As we are assuming that the fraction of malicious collectors overall does not exceed 1/3  (measured
+		//    by stake), the probability for randomly assigning 2/3 or more byzantine collectors to a single cluster
+		//    vanishes (provided a sufficiently high collector count in total).
```

Comment:
> In general the protocol needs to handle a Byzantine cluster.

I agree, but this is a very extreme edge case. Assuming close to 1/3 byzantine fraction in the overall collector population, it is reasonably likely to sample a cluster, whose local fraction of byzantine nodes exceeds 1/3. However, the case Tarak describes here requires that the cluster has 2/3 byzantine participants. The probabilities of this are described by the [Hypergeometric distribution](https://en.wikipedia.org/wiki/Hypergeometric_distribution). With multiple hundred collectors in the mature protocol and cluster sizes around 50 nodes or above, the probability of reaching 2/3 byzantine fraction in a cluster is very small.  For example, consider 500 collectors in total, 166 of which are byzantine. If we partition the collectors in clusters of size 50, the probability that one particular cluster reaches 2/3 byzantine members is `2 * 10^-8`. One the one hand, we would sample constitute 10 cluster each epoch, but an epoch is also 1 week long. 

I think it would be nice if we could handle this edge case with little extra work. But I think we could pragmatically also just disregard this edge case as "vanishing probability".

Reply:
>  it is reasonably likely to sample a cluster, whose local fraction of byzantine nodes exceeds 1/3. However, the case Tarak describes here requires that the cluster has 2/3 byzantine participants.

👍  I agree, I was conflating the "Byzantine-supermajority" case with "no honest supermajority" case - I'll resolve this